src/locales/context.js

```diff
@@ -0,0 +1,40 @@
+import { h, Component } from 'preact'
+import { createContext } from 'preact-context'
+import { initializeI18n } from './index.js'
+
+const LocaleContext = createContext()
+
+export class LocaleProvider extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      i18n: initializeI18n(props.language),
+    }
+  }
+
+  componentWillUpdate(nextProps) {
+    if (nextProps.language !== this.props.language) {
+      this.setState({
+        i18n: initializeI18n(nextProps.language)
+      })
+    }
+  }
+
+  render() {
+    const { i18n } = this.state
+    const { language, children } = this.props
+    const t = i18n.t.bind(i18n)
+    return (
+      <LocaleContext.Provider value={{ language, t }}>
+        {children}
+      </LocaleContext.Provider>
+    )
+  }
+}
+
+export const localised = WrappedComponent =>
```

Comment:
This looks good! Not sure if it's a good practice but I was thinking that we could have all the app's children localised by default and we can pass a variable for disabling localisation for children that don't need it. Just a thought

Reply:
Glad that you like it.

> have all the app's children localised by default
I think this could be a way to do so. Any child could consume `t` if they need to, only by being wrapped by the `localised` HOC. Are you concerned that having to call `localise(Component)` is introducing more clutter?

Reply:
Yes, that was my concern. A HOC seems like a good idea to me 👍